### DISCUSSION TITLE
how do I deploy a contract from a needed user with constructor args and sending ETH to the constructor?....

### DISCUSSION THREAD

**Discussion by moneyDev1111** [NomicFoundation/hardhat#4437] [General]
hi
how do I deploy a contract from a needed user with constructor args and sending ETH to the constructor?....
`await deployContract('aContract', [some.target], aNeededUser, { value: parseEther('1') });`
this of course gives a type error that there should be only 3 args.....
thanks

**Comment by fvictorio** [comment]
The second parameter is the array of constructor arguments.

**Reply by moneyDev1111** [reply]
the thing is that it's not possible to send eth to the constructor using deployContract() with a signer and constructor args because the method expects only 3 args, not 4

**Reply by fvictorio** [reply]
Ok, I think you mean this:

```js
await deployContract('aContract', [some.target], { value: parseEther('1'), signer: aNeededUser });
```

**Reply by moneyDev1111** [reply]
Thank you!:)
